2-repositories/iglu-server/src/main/scala/com/snowplowanalytics/iglu/server/Webhook.scala

```diff
@@ -2,29 +2,34 @@ package com.snowplowanalytics.iglu.server
 
 import cats.Applicative
 import cats.implicits._
-import com.snowplowanalytics.iglu.core.SchemaKey
+
 import io.circe.{Encoder, Json}
-import org.http4s.{Request, Uri}
+
+import org.http4s.{Request, Response, Status, Uri}
 import org.http4s.client.Client
 
+import com.snowplowanalytics.iglu.core.SchemaKey
+
 sealed trait Webhook
 
 object Webhook {
+
   case class SchemaPublished(uri: Uri, vendorPrefixes: Option[List[String]]) extends Webhook
-  case class ApiKeyPublished(uri: Uri, vendorPrefixes: Option[List[String]]) extends Webhook
 
   case class WebhookClient[F[_]](webhooks: List[Webhook], httpClient: Client[F]) {
-    def schemaPublished(schemaKey: SchemaKey, updated: Boolean)(implicit F: Applicative[F]): F[Unit] =
-      webhooks.traverse_ {
-        case SchemaPublished(uri, _) =>
+    def schemaPublished(schemaKey: SchemaKey, updated: Boolean)(implicit F: Applicative[F]): F[List[Either[String, Unit]]] =
+      webhooks.traverse {
+        case SchemaPublished(uri, prefixes) if prefixes.isEmpty || prefixes.getOrElse(List()).exists(schemaKey.vendor.startsWith(_)) =>
```

Comment:
I think the good way to avoid this very long and ugly line is to extract `prefixes` on as `List[String]`, but as `List[Vendor]`: https://github.com/snowplow/iglu/blob/iglu-server-ng/2-repositories/iglu-server/src/main/scala/com/snowplowanalytics/iglu/server/model/Permission.scala#L87